flang/runtime/io-api.cpp

```diff
@@ -784,6 +784,35 @@ bool IONAME(InputInteger)(Cookie cookie, std::int64_t &n, int kind) {
   return false;
 }
 
+bool IONAME(OutputReal32)(Cookie cookie, float x) {
+  IoStatementState &io{*cookie};
+  if (!io.get_if<OutputStatementState>()) {
+    io.GetIoErrorHandler().Crash(
+        "OutputReal32() called for a non-output I/O statement");
+    return false;
+  }
+  if (auto edit{io.GetNextDataEdit()}) {
+    return RealOutputEditing<24>{io, x}.Edit(*edit);
+  }
+  return false;
+}
+
+bool IONAME(InputReal32)(Cookie cookie, float &x) {
```

Comment:
Is InputReal32 defined in the header file? (I don't remember it.)

Reply:
Yep, I guess someone simply forgot to write the source function for it.

https://github.com/flang-compiler/f18-llvm-project/blob/fb701b69835a8e3dbf32f7e50c7874ad13391824/flang/runtime/io-api.h#L220-L221